### ISSUE TITLE
Investigate migrating documentation to Stardoc

### ISSUE THREAD

**Issue by jayconrod** [bazelbuild/rules_go#2437]
[Stardoc](https://github.com/bazelbuild/stardoc) is a documentation generator for Bazel rules, providers, and other definitions.

Currently, rules_go documentation is maintained separately from the rule implementation themselves, so it's easy for it to drift.

We should investigate whether Stardoc is a viable replacement for the current documentation. If it is, we should migrate the current documentation to generated Stardoc.

Analogous Gazelle issue bazelbuild/bazel-gazelle#763

**Comment by alexeagle** [comment]
The main question here is whether to switch to Markdown for this documentation, or write and maintain a custom velocity template so that stardoc can produce the reStructuredText matching the present docs.

**Comment by alexeagle** [comment]
I'm pretty sure we can show all the same content (markdown doesn't have rowspan in tables so the structure will be a bit different, but I think other rulesets have readable docs using markdown). If it were greenfield it would be obvious to use what Stardoc already knows how to write.

However if we change markup languages, we'll have a much harder time reviewing the giant delta to be sure all the docs are the same. With restructuredtext velocity templates we could more-or-less match the current content and just make it the generated output. The tricky bit remaining there is to make the columns align on the generated tables, which in rst actually do have to align as ascii art. I couldn't find an rst formatter that could automatically do that.

**Comment by alexeagle** [comment]
I don't think it should be closed yet, as there are still other .rst files containing manually-curated API docs
